torch/onnx/utils.py

```diff
@@ -483,8 +483,20 @@ def _model_to_graph(model, args, verbose=False,
                             module=module)
     from torch.onnx.symbolic_helper import _onnx_shape_inference
     if isinstance(model, torch.jit.ScriptModule) or isinstance(model, torch.jit.ScriptFunction):
-        assert example_outputs is not None, "example_outputs must be provided when exporting a ScriptModule or " \
-                                            "ScriptFunction."
+        input_args = copy.deepcopy(args)
+        input_kwargs = {}
+        if input_args and isinstance(input_args[-1], dict):
+            input_kwargs = input_args[-1]
+            input_args = input_args[:-1]
+        try:
+            model_copy = copy.deepcopy(model)
```

Comment:
why we need deepcopy model here? This seems expensive for big model. Can we just use `model` as below?

Reply:
My concern is that it may have the inplace ops in model. So we need deep copy this model.

Reply:
For inplace ops issue, we can the input value since it would change. Why we need copy the model?

Reply:
Sorry, I have added the deepcopy logical for inplace ops issue in [line 485](https://github.com/pytorch/pytorch/blob/d853f72a228c7ed61e3831549f35772854606680/torch/onnx/utils.py#L485).

The reason why we copy the model is for some cases weights or attributes changed after running the model once.

Reply:
I have tried to test the weight changed case. 
Such as batchnorm training mode, if we don't use deepcopy() to copy a new model, the weight `batch_model.running_mean` and `batch_model.running_var` will be changed. However, the unit tests will not be failed, cause we export a model with the new weight with `batch_model.running_mean` and `batch_model.running_var`, but we don't expect this behaviour.